### PR TITLE
week3 homework ahmet yasin

### PR DESCRIPTION
**Pull request by ahmtysn**

### CODE REVIEW

File: Week3/homeWork/creditCard/validator.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  // 3-) If sum is NaN, it's means there is an input for digits instead of number..
+  for (let i in strCardNums) {
+    sum += parseFloat(strCardNums[i]);
+    if (strCardNums[i] === strCardNums[0]) {
+      sameCount++;
+    }
+  }
+  if (strCardNums.length !== 16) {
+    statement = "Credit card number must be 16 digits..";
+  } else if (isNaN(sum)) {
```
Comment (eidosam) [inline Week3/homeWork/creditCard/validator.js]: That is smart!
But just to make it straight forward approach I'd use  for example `Array.from(strCardNums).some(digit => isNaN(digit))`

File: Week3/homeWork/creditCard/validator.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  return statement;
+}
+
+console.log(`Example 1: ${validator("1496320178541")}`);
+console.log(`Example 2: ${validator("14852hdkio124780")}`);
+console.log(`Example 3: ${validator("7512014785201439")}`);
+console.log(`Example 4: ${validator("1111111111110000")}`);
+console.log(`Example 5: ${validator("6666666666666666")}`);
+console.log(`Example 6: ${validator("4821059100451970")}`);
```
Comment (eidosam) [inline Week3/homeWork/creditCard/validator.js]: 🏅

File: Week3/homeWork/js-exercises/1-amazing.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "clever",
+    "super",
+    "hero",
+    "succesful",
+    "handsome",
+    "beautiful",
+    "faithful",
+    "hopeful"
+  ];
+  const randomTitle = titles[Math.floor(Math.random() * 10)];
```
Comment (eidosam) [inline Week3/homeWork/js-exercises/1-amazing.js]: 10 should be replaced with `titles.length`

File: Week3/homeWork/js-exercises/1-amazing.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  return `You are ${randomTitle}, ${name}!`;
+}
+
+console.log(giveCompliment("Noer"));
+console.log(giveCompliment("Wilgert"));
```
Comment (eidosam) [inline Week3/homeWork/js-exercises/1-amazing.js]: 👍

File: Week3/homeWork/js-exercises/2-dogYears.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+function calculateDogAge(age) {
+  const calculatedAge = age * 7;
+  if (typeof age !== "number") {
+    return "Please write value as a number...";
+  } else {
+    return `Your doggie is ${calculatedAge} years old in dog years!`;
+  }
+}
+
+console.log(calculateDogAge(13));
```
Comment (eidosam) [inline Week3/homeWork/js-exercises/2-dogYears.js]: 👍

File: Week3/homeWork/js-exercises/3-fortuneTeller.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    "California",
+    "Barcelona",
+    "Paris",
+    "Istanbul"
+  ];
+  const jobs = ["Pilot", "Engineer", "Teacher", "Performer", "Manager"];
+  return `You will be a ${jobs[randomNum]} in ${locations[randomNum]}, and married to ${partnerNames[randomNum]} with ${numChildren[randomNum]} kids.`;
+}
+
+console.log(tellFortune());
```
Comment (eidosam) [inline Week3/homeWork/js-exercises/3-fortuneTeller.js]: Arrays should be declared and initialized outside the function, and passed as arguments to the function.
Also I think you should generate a random number for each list, currently teacher would be always married to Marry and they have 3 kids and they live in Barcelona.

File: Week3/homeWork/js-exercises/4-shopping.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  listCard.push(item);
+  if (listCard.length > 3) {
+    listCard.shift();
+  }
+  return `You bought ${listCard.join(", ")}!`;
+}
+
+addToShoppingCart("cheese");
+addToShoppingCart("bread");
+console.log(addToShoppingCart("honey"));
```
Comment (eidosam) [inline Week3/homeWork/js-exercises/4-shopping.js]: 👍

File: Week3/homeWork/js-exercises/3-fortuneTeller.js (hunk 6 above)
Comment (ahmtysn) [reply Week3/homeWork/js-exercises/3-fortuneTeller.js]: i make a function for random numbers and problem was solved.
thanks:)

File: Week3/homeWork/js-exercises/1-amazing.js (hunk 3 above)
Comment (ahmtysn) [reply Week3/homeWork/js-exercises/1-amazing.js]: yes i changed..

File: Week3/homeWork/creditCard/validator.js (hunk 1 above)
Comment (ahmtysn) [reply Week3/homeWork/creditCard/validator.js]: Ooo it's awesome, thanks for your advice..

**Pull request closed without merging**